src/mlpack/methods/ann/ffn.hpp

```diff
@@ -440,6 +440,23 @@ class FFN
    */
   void ResetData(MatType predictors, MatType responses);
 
+ 
+   /**
+   * Quantize the neural network to use a different matrix type.
+   *
+   * This function creates a new neural network with the same structure
+   * but uses the specified target matrix type for its internal operations.
+   *
+   * @tparam TargetMatType The matrix type to quantize to.
+   * @return A new neural network with the specified target matrix type.
+   */
+  template<typename TargetMatType>
+  FFN<..., TargetMatType> Quantize() const
+  {
+    return ann::Quantize<MatType, TargetMatType>(*this);
+  }
```

Comment:
This is closer to the API I would hope for; but we should also allow specifying the quantization strategy as a template parameter. :+1: